sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-text.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from "@nestjs/common";
+import { plainToClass } from "class-transformer";
+import { validateSync } from "class-validator";
+import { flattenErrorMessages } from "../../utilities/class-validation";
+import { removeUTF8BOM } from "../../utilities";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+import {
+  ApplicationBulkWithdrawalData,
+  ApplicationBulkWithdrawalFooter,
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextModel,
+  ApplicationWithdrawalTextValidationResult,
+  RecordType,
+} from "./application-bulk-withdrawal-text.models";
+import { APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR } from "../../constants";
+import { CustomNamedError } from "@sims/utilities";
+
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationWithdrawalImportTextService {
+  /**
+   * Reads a text content and generate a text object model for each line.
+   * @param textContent text content to generate the models.
+   * @returns text object models.
+   */
+  readText(textContent: string): ApplicationWithdrawalTextModel[] {
+    const applicationWithdrawalModels: ApplicationWithdrawalTextModel[] = [];
+    // Remove BOM(Byte order mark), if present.
+    textContent = removeUTF8BOM(textContent);
+    const fileLines = textContent.split("\n");
+    // Read the first line to check if the header code is the expected one.
+    const header = ApplicationBulkWithdrawalHeader.createFromLine(
+      fileLines.shift(),
+    ); // Read and remove header.
+    if (
```

Comment:
This should not be an explicit validation like integration files, it must be part of class validator validations.